src/Pretty.hs

```diff
@@ -98,73 +110,74 @@ instance (PrettyBinder VarInfo typePat, PrettyBinder VarInfo pat, Pretty VarInfo
       Nothing -> string ("!!" ++ show v ++ "!!")
       Just (Stx _ _ x) -> text x
   pp env (CoreLet x@(Stx _ _ y) v def body) =
-    hang 2 $ group $
-    vsep [ text "let" <+> hang 2 (group (vsep [ pp env y <+> text "="
+    hang 2 $
+    vsep [ text "let" <+> hang 2 ((vsep [ pp env y <+> text "="
                                               , pp env def
                                               ])) <+> text "in"
          , pp (env <> Env.singleton v x ()) body
          ]
   pp env (CoreLetFun f@(Stx _ _ g) fv x@(Stx _ _ y) v def body) =
-    hang 2 $ group $
+    hang 2 $
     vsep [ text "flet" <+>
-           hang 2 (group (vsep [ pp env g <+> pp env y <+> text "="
+           hang 2 ((vsep [ pp env g <+> pp env y <+> text "="
                                , pp (env <> Env.singleton fv f () <> Env.singleton v x ()) def
                                ])) <+>
            text "in"
          , pp (env <> Env.singleton fv f ()) body
          ]
   pp env (CoreLam n@(Stx _ _ x) v body) =
-    hang 2 $ group $
+    hang 2 $
     text "λ" <> annotate (BindingSite v) (text x) <> "." <> line <>
     pp (env <> Env.singleton v n ()) body
   pp env (CoreApp fun arg) =
     hang 2 $ parens (pp env fun <> line <> pp env arg)
   pp env (CoreCtor ctor []) = pp env ctor
   pp env (CoreCtor ctor args) =
-    hang 2 $ parens $ pp env ctor <+> group (vsep (map (pp env) args))
+    hang 2 $ parens $ pp env ctor <+> (vsep (map (pp env) args))
   pp env (CoreDataCase _ scrut cases) =
-    hang 2 $ group $
+    hang 2 $
     vsep [ text "case" <+> pp env scrut <+> "of"
          , encloseSep (flatAlt mempty (text "{" <> space))
                       (flatAlt mempty (space <> text "}"))
                       (flatAlt mempty (space <> text ";" <> space)) $
            map (\(pat, rhs) ->
                    let (ppPat, env') = ppBind env pat
-                   in hang 2 $ group $
```

Comment:
It seems to me that this is a strange change. Shouldn't the LHS and RHS of the pattern be on one line if they fit?

That is,

```
case foo of
  0 |-> 2
  1 |-> 5
```
rather than
```
case foo of
  0 |->
    2
  1 |->
    5
```
?